### PR TITLE
Add ol4 support.

### PR DESCRIPTION
**Pull request by KaiVolland**
This target of this PullRequest is to support ol-versions >4 in our BasiGX components.

### CODE REVIEW

**Comment by marcjansen** [comment]
Very nice! Thanks.

* Did you test everything on ol 4 or is this the 'just' the stuff you stumbled upon?
* Will you also enable all of GeoExt for v4?
* Will this break backwards compatibility for clients that (still) work on ol 3.x

**Comment by KaiVolland** [comment]
In the current state it just fixed the buttons i needed. I tackle the rest later... not sure if I'm going for the geoext stuff, too.

**Comment by KaiVolland** [comment]
I think all breaking changes should now be catched. Plz review once more.

**Comment by KaiVolland** [comment]
@marcjansen 
Is it possible to merge this and refactor it later if the `isOl4`-check gets introduced to geoext3? This PR is somewhat blocking to me...

**Comment by marcjansen** [comment]
I'look into it tomorrow

File: src/plugin/Hover.js [hunk 1]
```diff
@@ -180,7 +180,7 @@ Ext.define('BasiGX.plugin.Hover', {
         }
     },
 
-   /**
+    /**
```
Comment (marcjansen) [inline src/plugin/Hover.js]: wrong autofix

File: src/util/Animate.js [hunk 2]
```diff
@@ -131,14 +131,14 @@ Ext.define('BasiGX.util.Animate', {
...
-             // tell OL3 to continue postcompose animation
+                // tell OL3 to continue postcompose animation
                 frameState.animate = true;
             }
             listenerKey = map.on('postcompose', animate);
             return listenerKey;
         },
 
-       /**
+        /**
```
Comment (marcjansen) [inline src/util/Animate.js]: wrong autofix

File: src/view/button/Measure.js [hunk 3]
```diff
@@ -249,6 +249,7 @@ Ext.define('BasiGX.view.button.Measure', {
 
         if (Ext.isEmpty(measureLayer)) {
             measureLayer = me.createMeasureLayer();
+            measureLayer.set('bp_displayInLayerSwitcher', false);
```
Comment (marcjansen) [inline src/view/button/Measure.js]: technically this is not related to the ol4 upgrade, is it?

File: src/util/Layer.js [hunk 4]
```diff
@@ -141,6 +141,9 @@ Ext.define('BasiGX.util.Layer', {
          * @return {Array} An array of all Layers.
          */
         getAllLayers: function(map) {
+            if (!map) {
+                map = BasiGX.util.Map.getMapComponent().getMap();
+            }
```
Comment (marcjansen) [inline src/util/Layer.js]: This isn't related to the upgrade, is it?

File: src/view/container/Redlining.js [hunk 5]
```diff
@@ -135,11 +135,11 @@ Ext.define('BasiGX.view.container.Redlining', {
      */
     stateChangeActive: false,
 
-   /**
+    /**
```
Comment (marcjansen) [inline src/view/container/Redlining.js]: wrong auto fix

File: src/view/container/Redlining.js [hunk 6]
```diff
@@ -135,11 +135,11 @@ Ext.define('BasiGX.view.container.Redlining', {
...
     *
     */
     config: {
-       /**
+        /**
```
Comment (marcjansen) [inline src/view/container/Redlining.js]: wrong autofix

File: src/view/container/WfsSearch.js [hunk 7]
```diff
@@ -52,6 +52,15 @@ Ext.define('BasiGX.view.container.WfsSearch', {
         }
     },
 
+    inheritableStatics: {
+        /**
+        * The name of the searchResultVectorLayer
+        *
+        * @type {String}
+        */
+        SEARCH_RESULT_VECTOR_LAYER_NAME: 'searchResultVectorLayer'
```
Comment (marcjansen) [inline src/view/container/WfsSearch.js]: comment is wrong indented, and this change is not related to the ol4 upgrade, right?

File: src/view/container/WfsSearch.js [hunk 8]
```diff
@@ -256,6 +265,7 @@ Ext.define('BasiGX.view.container.WfsSearch', {
      */
     initComponent: function() {
         var me = this;
+        var staticME = BasiGX.view.container.WfsSearch;
```
Comment (marcjansen) [inline src/view/container/WfsSearch.js]: `...ME` -> `...Me`

**Comment by KaiVolland** [comment]
I'll merge now.
